templates/raspbian_cattlepi/resources/usr/share/initramfs-tools/scripts/cattlepi-base/helpers

```diff
@@ -352,12 +352,42 @@ cattlepi_build_root_filesystem()
     cp "${CATTLEPI_DIR}/config" "${rootmnt}/cattlepi/"
     [ -r "${CATTLEPI_DIR}/apikey" ] && cp "${CATTLEPI_DIR}/apikey" "${rootmnt}/cattlepi/"
     [ -r "${CATTLEPI_DIR}/id" ] && cp "${CATTLEPI_DIR}/id" "${rootmnt}/cattlepi/"
+    # Copy the wireless info to the image!
+    [ -r "${CATTLEPI_DIR}/wpa_supplicant.conf" ] && cp "${CATTLEPI_DIR}/wpa_supplicant.conf" "${rootmnt}/etc/wpa_supplicant/wpa_supplicant.conf"
     echo $CATTLEPI_BASE > "${rootmnt}/cattlepi/base"
     echo "boot/${CATTLEPI_ID}/config" > "${rootmnt}/cattlepi/base_relative_config"
 }
 
+cattlepi_handle_wpa_supplicant() {
+    cattlepi_fetch_update_config
+
+    target_wpa_supplicant=$(jq -re ".wpa_supplicant" "${CATTLEPI_DIR}/config")
+    if [ $? -ne 0 ]; then
+        echo "No wpa_supplicant to handle"
+        return
+    fi
+
+    echo "${target_wpa_supplicant}" | base64 -d > /tmp/wpa_supplicant.conf
+    if [ $? -ne 0 ]; then
+        echo "Invalid or empty wpa_supplicant, skipping wireless"
+        return
+    fi
+
+    if [ -r "${CATTLEPI_DIR}/wpa_supplicant.conf" ]; then
+        cmp -s "${CATTLEPI_DIR}/wpa_supplicant.conf" /tmp/wpa_supplicant.conf
+        if [ $? -ne 0 ]; then
+            mv /tmp/wpa_supplicant.conf "${CATTLEPI_DIR}/wpa_supplicant.conf"
+            echo "wpa_supplicant.conf updated"
```

Comment:
should we force a reboot if the supplicant was written? this way we could use it immediately. 
also, one question i had was: if the initrd writes the supplicant, can the actual raspbian os in which we're booting use ti the first time? (or do we need a reboot also?)

Reply:
Hmm.  Yeah, I suppose we could reboot right away?

After wpa_supplicant is written to $CATTLEPI_DIR in this step, later in the boot process the build step runs which should copy the file to the raspbian filesystem in the appropriate place.